workflow/artifacts/s3/s3.go

```diff
@@ -103,6 +105,28 @@ func (s3Driver *S3ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifac
 				log.Warnf("Failed to test if %s is a directory: %v", path, err)
 				return false, nil
 			}
+
+			createBucketIfNotPresent := outputArtifact.S3.CreateBucketIfNotPresent
+			if createBucketIfNotPresent != nil && createBucketIfNotPresent.Enabled {
+				bucketExists, err := s3cli.BucketExists(outputArtifact.S3.Bucket)
```

Comment:
I suggest we just do a single `MakeBucket` and ignore duplicate errors. 50% of the network requests to go missing